src/runtime/metal.cpp

```diff
@@ -88,29 +88,23 @@ struct NSRange {
     size_t length;
 };
 
-WEAK void did_modify_range_if_supported(mtl_buffer *buffer, NSRange range) {
-  debug(NULL) << "did_modify_range_if_supported called.\n";
-    typedef bool (*responds_to_selector_method)(objc_id obj, objc_sel sel_1, objc_sel sel_2);
-    responds_to_selector_method method1 = (responds_to_selector_method)&objc_msgSend;
-    objc_sel did_modify_range_sel = sel_getUid("didModifyRange:");
-    if ((*method1)(buffer, sel_getUid("respondsToSelector:"), did_modify_range_sel)) {
-      debug(NULL) << "Sending didModifyRange method\n";
-        typedef void (*did_modify_range_method)(objc_id obj, objc_sel sel, NSRange range);
-        did_modify_range_method method2 = (did_modify_range_method)&objc_msgSend;
-        (*method2)(buffer, did_modify_range_sel, range);
-    }
+WEAK void did_modify_range(mtl_buffer *buffer, NSRange range) {
+    typedef void (*did_modify_range_method)(objc_id obj, objc_sel sel, NSRange range);
+    did_modify_range_method method = (did_modify_range_method)&objc_msgSend;
+    (*method)(buffer, sel_getUid("didModifyRange:"), range);
 }
 
-WEAK void synchronize_resource_if_supported(mtl_blit_command_encoder *encoder, mtl_buffer *buffer) {
-    typedef bool (*responds_to_selector_method)(objc_id obj, objc_sel sel_1, objc_sel sel_2);
-    responds_to_selector_method method1 = (responds_to_selector_method)&objc_msgSend;
-    objc_sel synchronize_resource_sel = sel_getUid("synchronizeResource:");
-    if ((*method1)(encoder, sel_getUid("respondsToSelector:"), synchronize_resource_sel)) {
-      debug(NULL) << "Sending synchronizeResource method\n";
-        typedef void (*synchronize_resource_method)(objc_id obj, objc_sel sel, mtl_buffer*buffer);
-        synchronize_resource_method method2 = (synchronize_resource_method)&objc_msgSend;
-        (*method2)(encoder, synchronize_resource_sel, buffer);
-    }
+WEAK void synchronize_resource(mtl_blit_command_encoder *encoder, mtl_buffer *buffer) {
+    typedef void (*synchronize_resource_method)(objc_id obj, objc_sel sel, mtl_buffer *buffer);
+    synchronize_resource_method method = (synchronize_resource_method)&objc_msgSend;
+    (*method)(encoder, sel_getUid("synchronizeResource:"), buffer);
+}
+
+WEAK bool is_buffer_managed(mtl_buffer *buffer) {
+    typedef int (*storage_mode_method)(objc_id obj, objc_sel sel);
```

Comment:
I believe the "storageMode" selector is only supported in iOS 9 and later. Thus this call needs to check if the method is supported before trying to call it. (The code will likely work as it stands because dispatching nonexistent methods returns nil, but depending on that is a bad idea and I believe it does log warning messages in certain execution contexts and possibly fails under stricter execution models or checker tools.) I think this is the only selector test still needed as if storageMode exists, all the other methods will exist as well.